packages/libs/src/sdk/middleware/addAppInfoMiddleware.ts

```diff
@@ -51,7 +60,7 @@ export const addAppNameMiddleware = ({
         url:
           context.url +
           (context.url.includes('?') ? '&' : '?') +
-          querystring({ app_name: appName ?? '' }),
+          querystring({ app_name: appName ?? '', api_key: apiKey! }),
```

Comment:
Eventually this should be a basic auth header instead probably

Reply:
in that case I think the `addRequestSignatureMiddleware` and this one need to become one so that they can duke out who goes there

Reply:
vague ugly pseudo code
```ts
try {
  return addSignedAuthHeader(...)
} catch (e) {
  if (e instanceof NotImplementedError) {
    try {
      return addPublicKeyAuthHeader(...)
    } catch (err) {
      if (err instanceof NotImpelmentedError) {
        return addAppName()
      }
    }
  }
```